src/models/progress/useProgressStateQuery.js

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useMemo, useState } from 'react';
+import { QueryObserver, useQueryClient } from 'react-query';
+import { getProgressQueryKey } from '../../constants/queryKeys';
+import { progressState } from '../../constants/progress';
+import {
+  isProgressRejected,
+  isProgressResolved,
+} from '../../utils/progressUtils';
+import {
+  getProgress,
+  queryOptions as defaultQueryOptions,
+} from './progressQueryDefaults';
+
+const defaultState = {};
+
+export default function useProgressStateQuery(progressGuid, enabled) {
+  const [state, setState] = useState(defaultState);
+  const queryClient = useQueryClient();
+  const queryObserver = useMemo(
+    () => {
+      if (!progressGuid) return null;
+
+      const observerOptions = {
+        ...defaultQueryOptions,
+        queryKey: getProgressQueryKey(progressGuid),
+        queryFn: getProgress,
+        notifyOnPropsChange: ['data', 'error', 'isFetched'],
+      };
+
+      if (enabled !== undefined) observerOptions.enabled = enabled;
```

Comment:
what if enabled were `false`? i'm confused about why you're only looking for undefined here. what about like `enabled: Boolean(enabled)`?

Reply:
If I don't pass in enabled, I don't want to set that property on the options. That way it can use whatever the query's default value is.